loader/include/Geode/ui/ProgressBar.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <cocos2d.h>
+
+namespace geode {
+    // Enum for progress bar style
+    enum class ProgressBarStyle {
+        Level = 0,
+        Solid = 1,
+    };
+
+    // Custom class for the progress bar
+    class ProgressBar : public cocos2d::CCNode {
+    protected:
+        class Impl;
+        std::unique_ptr<Impl> m_impl;
+
+        ProgressBar();
+        ~ProgressBar();
+
+        void customSetup();
```

Comment:
I tend to call functions like this `reload` or `reloadStyle` instead to make it clear that it may (and should) be called multiple times